test/test_tree_2d_euler.jl

```diff
@@ -235,6 +235,38 @@ end
     end
 end
 
+@trixi_testset "elixir_euler_shockcapturing_subcell.jl (fixed time step)" begin
```

Comment:
Would this test fail without your fixes?

Reply:
Yes, the fixed timestep would be completely ignored. There where differences when switching the tests from 1.9 to 1.10 (as for other integrators) but I could not make sure that the values where almost identical when using the same fixed timestep. This is how we found this bug

Reply:
Yes. As explained [above](https://github.com/trixi-framework/Trixi.jl/pull/1919#issuecomment-2085547384), before, the time step would just be `0.1+1.0e-8` in this setting and the simulation would crash immediately.

Reply:
Thanks

Reply:
Wait sorry. Give me a second

Reply:
The "new" save solution callback wasn't even considered before. Had to move it into the test itself.
Additionally, I adapted the fixed step size such that `0.1` is met exactly (Then, the test is clearer.)
Now I'm happy.